src/core/expr.rs

```diff
@@ -39,6 +41,62 @@ impl From<Op> for utils::SExp {
     }
 }
 
+/// A custom function definition
+pub struct FnAssume
+{
+    assumptions: Vec<Implication<Expr>>
+}
+
+impl FnAssume {
+    /// Creates a unique ID to be replaced for function definitions.
+    /// Returns `(l, r)`: [core::Id] terms for the left and right expressions used in `assume`
+    fn replaceable_ids() -> &'static (Id, Id) {
```

Comment:
Does this exist to allow us to have "pattern variables" in our axioms? For example, if we want to say:
```
x + 1 - 1 = x
```
It'll internally be represented as?
```
_FnAssume_left + 1 - 1 = x
```

Reply:
Yes, you would just use left on both sides and then bind left to x